promquery/query_collector.go

```diff
@@ -0,0 +1,123 @@
+package promquery
+
+import (
+	"fmt"
+	"strings"
+
+	"github.com/prometheus/client_golang/prometheus"
+)
+
+// QueryCollector is responsible for concentrate all metrics avaiable
+type QueryCollector struct {
+	totalCalls    *prometheus.CounterVec
+	totalDuration *prometheus.CounterVec
+	totalSuccess  *prometheus.CounterVec
+	totalFailures *prometheus.CounterVec
+}
+
+// QueryCollectorOpts is the input option for QueryCollector
+type QueryCollectorOpts struct {
+	// Prefix: responsible for all counters descs prefix
+	Prefix string
+}
+
+var queryCollectorLabels = []string{"queryName"}
+
+// NewQueryCollector returns a new QueryCollector pointer
+func NewQueryCollector(opts *QueryCollectorOpts) *QueryCollector {
+	prefix := opts.Prefix
+	if prefix != "" && !strings.HasSuffix(opts.Prefix, "_") {
+		prefix += "_"
+	}
+
+	// TODO: Add prefix name and descriptions
+	return &QueryCollector{
+		totalCalls: prometheus.NewCounterVec(
+			prometheus.CounterOpts{
+				Name: fmt.Sprintf("db_%stotal_calls", prefix),
```

Comment:
```suggestion
				Name: fmt.Sprintf("namedqry_%stotal_calls", prefix),
```

Also, apply it to the other metrics.